app/controllers/subscriptions_controller.rb

```diff
@@ -5,19 +5,24 @@ class SubscriptionsController < ApplicationController
 
   def create
     @subscription = current_user.subscriptions.build(conference_id: @conference.id)
-    if @subscription.save!
+    if @subscription.save
       redirect_to root_path, notice: "You have been subscribed to receive email notifications for #{@conference.short_title}."
     else
-      redirect_to root_path, error: subscription.errors.full_messages.to_sentence
+      redirect_to root_path, error: @subscription.errors.full_messages.to_sentence
     end
   end
 
   def destroy
     @subscription = current_user.subscriptions.find_by(conference_id: @conference.id)
-    if @subscription.destroy
+
+    if !@subscription
```

Comment:
You can extract this part into a standalone statement, so that it is not part of the if statement.
redirect_to ..., error: '...' unless @subscription

You can also redirect directly to root_path, that's the only place subscription comes from anyway.